### DISCUSSION TITLE
Neptune 4 Series Revo Rapid change Mod

### DISCUSSION THREAD

**Discussion by chuckkay** [OpenNeptune3D/OpenNept4une#62] [Ideas]
Now you can have the convenience of cold swapping a wide variety of nozzle sizes flows and material with this Mod I created.  Enjoy…. 
![IMG_0923](https://github.com/OpenNeptune3D/OpenNept4une/assets/10617746/00d78a68-a11f-45bc-ac85-19ed2be295dd)
![IMG_0922](https://github.com/OpenNeptune3D/OpenNept4une/assets/10617746/f5862a8d-77b6-42df-9d35-4666a1c18e17)
![IMG_0911](https://github.com/OpenNeptune3D/OpenNept4une/assets/10617746/e77b7029-9d4a-4d5f-b7ca-53bf1261e890)

Parts List for Elegoo Neptune series Revo Rapid change upgrade Mod

Neptune 4 & Neptune 4 Pro:

1 [Revo 24v 60w ceramic core kit](https://3dprintingusa.com/products/official-e3d-revo%E2%84%A2-heatercore-with-spring-24v-60w?variant=40602888273989). This is a great deal for an official at $48, if not available just search elsewhere, FYI there are cheaper knockoffs but be warned they struggle to maintain a stable temp, and will crash klipper at higher temperatures over 230c if the aux fan is on, also most knockoffs don’t come with the required Spring, so if you get a knockoff, you will need [1 Revo spring](https://3dprintingusa.com/products/official-e3d-revo%E2%84%A2-spring-triple-pack?variant=39987136856133&gclid=Cj0KCQiAn-2tBhDVARIsAGmStVkA-RMNgyo_5jLoNGyBz6KMH-3u6g2PsSQBKokL-ZjuO-bK3mC1m8EaAj6FEALw_wcB) (normally sold in 3 packs). Alternatively, you can salvage a spring from the AA battery compartment of most toys or electronics—just clip off the narrow part so it’s approximately the size of a Revo spring it will do the job of keeping the heater core in the right place, but considering its not the official size spring it will fall off whenever you remove the nozzle, official Revo spring clamp on to the heater core.

1 [Aluminum or copper sleeve M4 nut 20mm long 7mm outer dimension](https://a.co/d/gX7ZrdE) ***UPDATE*** these 7mm OD, are nearly impossible to find, but a 22mm long, 6mm OD version are commonly available on Amazon https://a.co/d/5NB5xQM and you just need to wrap it a couple of times with aluminum heat tape and a coating of thermal paste, and its just as good.. just be sure to tighten the grub screws real good or the sleeve will fall out, another option is you can use an M4 heat set insert which is easier to obtain but much shorter, thus less secure, less thermal, but works. 

The following parts from a [Neptune 4 /4 pro hotend kit ](https://a.co/d/icL8pzi)or salvage them from your existing toolhead:
The Heatsink 
The blue PTFE TUBE cut it down to 28mm
The small grub screw thats in the heatsink which will be used for locking the M4 sleeve or heat set insert in place.
The heater Wires with XH2.54 2PIN male connector
The thermistor Wires with unknown model  2PIN male connector,

Neptune 4 Plus & Neptune 4 Max:

1 [Revo 24v 60w ceramic core kit](https://a.co/d/gJ7Q8Lt)

1 [revo spring](https://3dprintingusa.com/products/official-e3d-revo%E2%84%A2-spring-triple-pack?variant=39987136856133&gclid=Cj0KCQiAn-2tBhDVARIsAGmStVkA-RMNgyo_5jLoNGyBz6KMH-3u6g2PsSQBKokL-ZjuO-bK3mC1m8EaAj6FEALw_wcB) hard to find and normally sold in 3 packs ( you can salvage a spring from AA battery compartment of most toys or electronics - just clip off narrow part)

1 [Aluminum or copper sleeve M4 nut 20mm long 7mm outer dimension](https://a.co/d/gX7ZrdE) or you can use an M4 heat set insert which is easier to obtain but less secure, less thermal, but works. 

The following parts from  a [Neptune 4 /4 pro hotend kit ](https://a.co/d/icL8pzi)  (yes i said Neptune 4 /4 pro hotend kit because the plus/max heatsink in is too short and the heatbreak hole is too small for this mod):
The Heatsink 
The blue PTFE TUBE cut it down to 28mm
The small grub screw thats in the heatsink which will be used for locking the M4 sleeve or heat set insert in place.

The following wire connectors from your existing thermistor and heater that connect to your toolhead or buy a [Neptune 4 Plus/4 max hotend kit](https://a.co/d/dz2D7Wz) 
The heater Wires with XH2.54 2PIN male connector
The thermistor Wires with unknown model 2PIN male connector.

Installation:

Remove toolhead from gantry
Remove heater and thermistor wires from toolhead jumpers 
Remove 2 screws holding heatsink in place
Loosen grub screw from heatsink 
Remove bimetal heatbreak from heatsink
Take a neptune 4/4pro heatsink not a plus or max heatsink (too short) Insert either the 20mm M4 sleeve nut or the M4 heat set insert, whichever you use let it stick out about 3mm, and then secure it by tightening the grub screw.
Place spring on revo ceramic core and try to snap fit it to hold on
Slide revo nozzle through heater core and spring and begin to screw it into heatsink. use JB weld or some other high heat rated metal glue to secure the other end of the spring to the heatsink, being careful not to get the glue on anything but the bottom of the heatsink and spring..

Now for the hard part, cut the manufacturers black molex 3.0 jumpers off the wires of the the revo heat core kit and splice on your wires with connectors that fit your toolhead thermistor and heater jumpers 

Reassemble everything, hopefully you did a neat job splicing as chunky splice or excessive wire length will make it difficult to reassemble the tool head and make everything fit..

Turn on your printer
Adjust your new z offset, as the revo nozzle maybe higher or lower then your original elegoo nozzle was previously.

change or add sensor_type in Klipper Printer.cfg file
In the [extruder] section 
sensor_type:ATC Semitec 104NT-4-R025H42G

Do a PID tuning to calibrate the new ceramic heater core.

Do a bed leveling..

Not necessary but highly recommended to convert your Neptune 4 to full Klipper, by de-Elegooizing it….  Follow the instructions here. https://github.com/OpenNeptune3D/OpenNept4une

**Comment by phillipthelen** [comment]
that's really cool! Would you mind if I copy this guide to the wiki? I think that's the best place to store guides like this long term :)

**Comment by chuckkay** [comment]
Sure no problem.

Question how do i switch to mainsail instead of fluid?
I have both running but i need to go to port 81 for mainsail.
Can i just uninstall fluid?

Also I see mainsail.cfg is commented out in printer.cfg, not sure how that works

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by Hoorow** [reply]
not sure but you can install mainsail without removing fluiid then change printer.cfg probably gonna need some tweak, if anything bad happens you can just re flash firmware

**Reply by halfmanbear** [reply]
Uninstall fluidd via kiauh then edit the port (command below) from 81 to 80, save and exit.

```
sudo nano /etc/nginx/sites-available/mainsail
```

**Comment by crazyslicster** [comment]
> Now you can have the convenience of cold swapping a wide variety of nozzle sizes flows and material with this Mod I created. Enjoy…. ![IMG_0923](https://private-user-images.githubusercontent.com/10617746/301729139-00d78a68-a11f-45bc-ac85-19ed2be295dd.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAxNzYxMDUsIm5iZiI6MTcxMDE3NTgwNSwicGF0aCI6Ii8xMDYxNzc0Ni8zMDE3MjkxMzktMDBkNzhhNjgtYTExZi00NWJjLWFjODUtMTllZDJiZTI5NWRkLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxMVQxNjUwMDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xYjkwYmRkODY5M2ViYzhkMTA3NmI2ZGQxZWQ0MTdjOGZjMzFjZmY3Yjg4YTk2YTRmNDlkMjU0ZDFiN2UxYWVhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.IxZclPWuTB5c9cZPpNvbXWbFPV9o_-tDhsPjrvqZcks) ![IMG_0922](https://private-user-images.githubusercontent.com/10617746/301729143-f5862a8d-77b6-42df-9d35-4666a1c18e17.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAxNzYxMDUsIm5iZiI6MTcxMDE3NTgwNSwicGF0aCI6Ii8xMDYxNzc0Ni8zMDE3MjkxNDMtZjU4NjJhOGQtNzdiNi00MmRmLTlkMzUtNDY2NmExYzE4ZTE3LmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxMVQxNjUwMDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZTBjMWUwMWY5ZWQxZTg3ZjEzMDA4MjljYTkzYjM2MGM0ZGYyNWI5YTQzOTAxYTg0MTBmMmUyM2ZjYjMyNzhmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.LxebQiGSg1PT0BcBJjJuzsMik9Hj2Gkbv60-oSToieo) ![IMG_0911](https://private-user-images.githubusercontent.com/10617746/301729146-e77b7029-9d4a-4d5f-b7ca-53bf1261e890.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAxNzYxMDUsIm5iZiI6MTcxMDE3NTgwNSwicGF0aCI6Ii8xMDYxNzc0Ni8zMDE3MjkxNDYtZTc3YjcwMjktOWQ0YS00ZDVmLWI3Y2EtNTNiZjEyNjFlODkwLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxMVQxNjUwMDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YTUxYzcxZjBlZGJlNWFmOWFhZjNkZjMyZWU0NDdlMmZkYjUyMjRlNWU4MzUyNjkzOTdmYzRiNGMyMTJkY2NmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Jl8JBPNdMx1mLSruM_JaZG4DO_3ao_EmDiLw57nFE1M)
> 
> Parts List for Elegoo Neptune series Revo Rapid change upgrade Mod
> 
> Neptune 4 & Neptune 4 Pro:
> 
> 1 [Revo 24v 60w ceramic core kit](https://3dprintingusa.com/products/official-e3d-revo%E2%84%A2-heatercore-with-spring-24v-60w?variant=40602888273989). This is a great deal for an official at $48, if not available just search elsewhere, FYI there are cheaper knockoffs but be warned they struggle to maintain a stable temp, and will crash klipper at higher temperatures over 230c if the aux fan is on, also most knockoffs don’t come with the required Spring, so if you get a knockoff, you will need [1 Revo spring](https://3dprintingusa.com/products/official-e3d-revo%E2%84%A2-spring-triple-pack?variant=39987136856133&gclid=Cj0KCQiAn-2tBhDVARIsAGmStVkA-RMNgyo_5jLoNGyBz6KMH-3u6g2PsSQBKokL-ZjuO-bK3mC1m8EaAj6FEALw_wcB) (normally sold in 3 packs). Alternatively, you can salvage a spring from the AA battery compartment of most toys or electronics—just clip off the narrow part so it’s approximately the size of a Revo spring it will do the job of keeping the heater core in the right place, but considering its not the official size spring it will fall off whenever you remove the nozzle, official Revo spring clamp on to the heater core.
> 
> 1 [Aluminum or copper sleeve M4 nut 20mm long 7mm outer dimension](https://a.co/d/gX7ZrdE) _**UPDATE**_ these 7mm OD, are nearly impossible to find, but a 22mm long, 6mm OD version are commonly available on Amazon https://a.co/d/5NB5xQM and you just need to wrap it a couple of times with aluminum heat tape and a coating of thermal paste, and its just as good.. just be sure to tighten the grub screws real good or the sleeve will fall out, another option is you can use an M4 heat set insert which is easier to obtain but much shorter, thus less secure, less thermal, but works.
> 
> The following parts from a [Neptune 4 /4 pro hotend kit ](https://a.co/d/icL8pzi)or salvage them from your existing toolhead: The Heatsink The blue PTFE TUBE cut it down to 28mm The small grub screw thats in the heatsink which will be used for locking the M4 sleeve or heat set insert in place. The heater Wires with XH2.54 2PIN male connector The thermistor Wires with unknown model 2PIN male connector,
> 
> Neptune 4 Plus & Neptune 4 Max:
> 
> 1 [Revo 24v 60w ceramic core kit](https://a.co/d/gJ7Q8Lt)
> 
> 1 [revo spring](https://3dprintingusa.com/products/official-e3d-revo%E2%84%A2-spring-triple-pack?variant=39987136856133&gclid=Cj0KCQiAn-2tBhDVARIsAGmStVkA-RMNgyo_5jLoNGyBz6KMH-3u6g2PsSQBKokL-ZjuO-bK3mC1m8EaAj6FEALw_wcB) hard to find and normally sold in 3 packs ( you can salvage a spring from AA battery compartment of most toys or electronics - just clip off narrow part)
> 
> 1 [Aluminum or copper sleeve M4 nut 20mm long 7mm outer dimension](https://a.co/d/gX7ZrdE) or you can use an M4 heat set insert which is easier to obtain but less secure, less thermal, but works.
> 
> The following parts from a [Neptune 4 /4 pro hotend kit ](https://a.co/d/icL8pzi) (yes i said Neptune 4 /4 pro hotend kit because the plus/max heatsink in is too short and the heatbreak hole is too small for this mod): The Heatsink The blue PTFE TUBE cut it down to 28mm The small grub screw thats in the heatsink which will be used for locking the M4 sleeve or heat set insert in place.
> 
> The following wire connectors from your existing thermistor and heater that connect to your toolhead or buy a [Neptune 4 Plus/4 max hotend kit](https://a.co/d/dz2D7Wz) The heater Wires with XH2.54 2PIN male connector The thermistor Wires with unknown model 2PIN male connector.
> 
> Installation:
> 
> Remove toolhead from gantry Remove heater and thermistor wires from toolhead jumpers Remove 2 screws holding heatsink in place Loosen grub screw from heatsink Remove bimetal heatbreak from heatsink Take a neptune 4/4pro heatsink not a plus or max heatsink (too short) Insert either the 20mm M4 sleeve nut or the M4 heat set insert, whichever you use let it stick out about 3mm, and then secure it by tightening the grub screw. Place spring on revo ceramic core and try to snap fit it to hold on Slide revo nozzle through heater core and spring and begin to screw it into heatsink. use JB weld or some other high heat rated metal glue to secure the other end of the spring to the heatsink, being careful not to get the glue on anything but the bottom of the heatsink and spring..
> 
> Now for the hard part, cut the manufacturers black molex 3.0 jumpers off the wires of the the revo heat core kit and splice on your wires with connectors that fit your toolhead thermistor and heater jumpers
> 
> Reassemble everything, hopefully you did a neat job splicing as chunky splice or excessive wire length will make it difficult to reassemble the tool head and make everything fit..
> 
> Turn on your printer Adjust your new z offset, as the revo nozzle maybe higher or lower then your original elegoo nozzle was previously.
> 
> change or add sensor_type in Klipper Printer.cfg file In the [extruder] section sensor_type:ATC Semitec 104NT-4-R025H42G
> 
> Do a PID tuning to calibrate the new ceramic heater core.
> 
> Do a bed leveling..
> 
> Not necessary but highly recommended to convert your Neptune 4 to full Klipper, by de-Elegooizing it…. Follow the instructions here. https://github.com/OpenNeptune3D/OpenNept4une

Do you know if the pro has the same parts as the Plus? I'd like to do this mod if the hardware is the same or similar? I live changing nozzles for bigger prints and heating + unscrewing is annoying. 

Thanks.

**Reply by chuckkay** [reply]
Your in luck, technically if your willing to reuse the existing thermistor wire and heater wire, then you don't need to buy the neptune 4/4pro hotend kit because your existing heatsink is the right size... but if you don't want to ruin your original then yes buy the 4/4pro hotend kit... and the other parts listed...  the hardest part to find is the 7mm Outer Diamiter M4 threaded sleeve, most available on amazon or 6mm OD M4 and that is not ideal but still doable

**Reply by chuckkay** [reply]
FYI I am absolutely loving my Neptune 4 Revo Max

**Reply by HFleischman265** [reply]
When I learned about screws_tilt, I swapped my center bed spacers out for springs.  It turns out that the original center spacer is a perfect adapter for the 6mm aluminum sleeve from amazon! 
![unnamed](https://github.com/OpenNeptune3D/OpenNept4une/assets/138933885/7e6c3a03-616b-4dfc-bb80-ca39352543fa)

**Reply by HFleischman265** [reply]
Harbor Freight has a fairly inexpensive spring assortment with a suitable spring also.

![Image_20240330_2208371132997145840618070](https://github.com/OpenNeptune3D/OpenNept4une/assets/138933885/2e24a659-79e3-48ff-99ea-b102349b2606)

**Reply by chuckkay** [reply]
> When I learned about screws_tilt, I swapped my center bed spacers out for springs.  It turns out that the original center spacer is a perfect adapter for the 6mm aluminum sleeve from amazon! 
> 
> ![unnamed](https://github.com/OpenNeptune3D/OpenNept4une/assets/138933885/7e6c3a03-616b-4dfc-bb80-ca39352543fa)
> 
> 

Nice find, nice repurposing..  I was using 6mm and aluminum tape, the 7mm OD took a month to arrive from china, but fit like a charm

**Comment by Ithriveontacos** [comment]
I love this guide and have been searching for something like this for weeks.

**Reply by chuckkay** [reply]
> I love this guide and have been searching for something like this for weeks. 

Glad to here it, i tried to cover everything in the guide...  FYI it's been 3 months now with my Neptune revo mod...and im happy to report zero problems.  I just want to install a wire brush and gcode to wipe the nozzle clean, as sometimes a little leftover filament  on the nozzle can mess up the adaptive mesh

**Comment by jDavidnet** [comment]
How do we like this mod in 2025?

**Comment by PJMCL1618033** [comment]
I'm going to try this mod on a NP4. 
I did a conversion that adapts a Bambu Lab P1S nozzle to fit the Pro 4. It required a remake of the hot end housing. 

McMASTER-CARR has some items that could be used for the throat. They are 8mm diameter OD so the heat sink would have to be enlarged. Or I might use an existing NP4 throat and drill & tap it.

Round Coupling Nut
18-8 Stainless Steel, M4 x 0.7 mm Thread, 20 mm Long
https://www.mcmaster.com/90202A101/


18-8 Stainless Steel Unthreaded Spacer
8 mm OD, 20 mm Long, for M3 Screw Size
https://www.mcmaster.com/92871A061/


Aluminum Unthreaded Spacer
8 mm OD, 20 mm Long, for M3 Screw Size
https://www.mcmaster.com/94669A061/
